src/alertlogic_lib/core.clj

```
@@ -93,14 +96,18 @@
   (integer string)."
   [customer-id api-token]
   (if (nil? customer-id)
-    (error "Customer ID cannot be nil. Aborting.")
+    (do
+      (error "Customer ID cannot be nil. Aborting.")
+      (md/success-deferred []))
     (let [url (str base-url-public (format lm-hosts-api customer-id))
-          hosts (:hosts @(get-page! url api-token))
           cleanup-host
-          (fn [host]
-            (let [{{:keys [name status metadata type]} :host} host]
-              {:name name
-               :status (:status status)
-               :ips (:local-ipv-4 metadata)
-               :type type}))]
-      (map cleanup-host hosts))))
+          (fn [{:keys [host]}]
+            (let [{:keys [status metadata]} host]
+              (merge
+               (rename-keys host {:status :status-data})
+               {:status (:status status)
+                :ips (:local-ipv-4 metadata)})))]
+      (md/chain
+       (get-page! url api-token)
```

The implementation of `get-page!` swallows exceptions and instead returns `::fetch-error` when it encounters any error. It doesn't look like any consumers explicitly handle this case, which makes me think that it might be difficult to understand what could go wrong in the event of a fetch failing. In the poller lib, I ended up removing fetch error and allowing it to return error deferreds, which made it much easier to use `catch`. Have you thought about this? Were you wanting to address error handling in another PR?